docs/test-development.md

```diff
@@ -0,0 +1,23 @@
+## Writing tests
+
+We focus primarily on integration/functional level tests. Unit tests are avoided and only recommended for isolated
+classes such as small utils. We do not use any mocking frameworks nor will be accept any contributions that uses these.
```

Comment:
"not will accept any...."

Reply:
"that use these" ? (Really not sure about this)

Reply:
Changed